xapian-core/weight/bm25weight.cc

```diff
@@ -103,11 +120,16 @@ BM25Weight::init(double factor)
 	}
     }
 #else
-    if (tw < 2) tw = tw * 0.5 + 1;
-    termweight = log(tw) * factor;
-    if (param_k3 != 0) {
-	double wqf_double = get_wqf();
-	termweight *= (param_k3 + 1) * wqf_double / (param_k3 + wqf_double);
+    if (tw < 2 && !bm25+) {
+        tw = tw * 0.5 + 1;
+        termweight = log(tw) * factor;
+        if (param_k3 != 0) {
+            double wqf_double = get_wqf();
+            termweight *= (param_k3 + 1) * wqf_double / (param_k3 + wqf_double);
+        }
+    }
+    else if (bm25+) {
```

Comment:
I'm not convinced this does the right thing for non-BM25+. If `tw >= 2` then `termweight` isn't set, while previously it would have been `log(tw) * factor`. What's probably simplest is an outer `if (bm25_plus) { termweight = log(tw) * factor; } else { … }` with the existing code inside.